File: src/types/character.ts
import { DefaultOptionType } from 'antd/es/select'

/**
 * 角色性别
 */
export enum CharacterGender {
  Male = 'male',
  Female = 'female',
  Other = 'other',
  Unknown = 'unknown',
}

/**
 * 角色性别标签
 */
export const CharacterGenderLabels: Record<CharacterGender, string> = {
  [CharacterGender.Male]: '男',
  [CharacterGender.Female]: '女',
  [CharacterGender.Other]: '其他',
  [CharacterGender.Unknown]: '未知',
}

/**
 * 角色选项
 */
export const CharacterGenderOptions: DefaultOptionType[] = [
  { value: CharacterGender.Male, label: CharacterGenderLabels[CharacterGender.Male] },
  { value: CharacterGender.Female, label: CharacterGenderLabels[CharacterGender.Female] },
  { value: CharacterGender.Other, label: CharacterGenderLabels[CharacterGender.Other] },
  { value: CharacterGender.Unknown, label: CharacterGenderLabels[CharacterGender.Unknown] },
]

/**
 * 角色类型
 */
export enum CharacterType {
  /** 主角 */
  Protagonist = 'protagonist',
  /** 二号主角 */
  SecondProtagonist = 'second_protagonist',
  /** 三号主角 */
  ThirdProtagonist = 'third_protagonist',
  /** 配角 */
  Supporting = 'supporting',
  /** 次要配角 */
  MinorSupporting = 'minor_supporting',
}

/**
 * 角色类型标签
 */
export const CharacterTypeLabels: Record<CharacterType, string> = {
  [CharacterType.Protagonist]: '主角',
  [CharacterType.SecondProtagonist]: '二号主角',
  [CharacterType.ThirdProtagonist]: '三号主角',
  [CharacterType.Supporting]: '配角',
  [CharacterType.MinorSupporting]: '次要配角',
}

/**
 * 角色类型选项
 */
export const CharacterTypeOptions: DefaultOptionType[] = [
  { value: CharacterType.Protagonist, label: CharacterTypeLabels[CharacterType.Protagonist] },
  {
    value: CharacterType.SecondProtagonist,
    label: CharacterTypeLabels[CharacterType.SecondProtagonist],
  },
  {
    value: CharacterType.ThirdProtagonist,
    label: CharacterTypeLabels[CharacterType.ThirdProtagonist],
  },
  { value: CharacterType.Supporting, label: CharacterTypeLabels[CharacterType.Supporting] },
  {
    value: CharacterType.MinorSupporting,
    label: CharacterTypeLabels[CharacterType.MinorSupporting],
  },
]

/**
 * 角色
 */
export interface Character {
  id: string
  name: string
  gender: CharacterGender
  characterType?: CharacterType | null
  novelId: string
  background?: string
  appearance?: string
  personality?: string
  additionalInfo?: string
  createdAt: string
  updatedAt: string
}

File: src-tauri/src/database/models/character.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::Type;
use uuid::Uuid;

use crate::string_enum;

// 使用宏定义性别枚举
string_enum! {
    /// 角色性别枚举
    #[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Type)]
    #[serde(rename_all = "snake_case")]
    #[sqlx(type_name = "TEXT", rename_all = "snake_case")]
    pub enum Gender {
        Male,     // 男
        Female,   // 女
        Other,    // 其他
        Unknown,  // 未知
    }
}

/// 角色实体模型
/// 对应数据库中的 characters 表，存储小说中的角色信息
#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct Character {
    pub id: Uuid,
    pub novel_id: Uuid,
    pub name: String,
    pub gender: Gender,
    pub background: Option<String>,
    pub appearance: Option<String>,
    pub personality: Option<String>,
    pub additional_info: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
